### PR TITLE
Update required Helm chart version to match APIGW release notes

### PR DESCRIPTION
**Pull request by nathancoleman**
### Description
The [README](https://github.com/hashicorp/consul-api-gateway/blob/main/README.md) and [Release Notes](https://developer.hashicorp.com/consul/docs/release-notes/consul-api-gateway/v0_4_x#supported-software) for Consul API Gateway v0.4 indicate that v0.47.1+ of the Consul Helm chart is required; however, it seems we missed updating the Tech Specs page.

### Links
See README and Release Notes links above

### PR Checklist

* [ ] updated test coverage
* [x] external facing docs updated
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/api-gateway/tech-specs.mdx [hunk 1]
```diff
@@ -17,12 +17,8 @@ Verify that your environment meets the following requirements prior to using Con
 
 Your datacenter must meet the following requirements prior to configuring the Consul API Gateway:
 
-- Kubernetes 1.21+
-  - Kubernetes 1.24 is not supported at this time.
-- `kubectl` 1.21+
-- Consul 1.11.2+
-- HashiCorp Consul Helm chart 0.45.0+
-- Consul Service Mesh must be deployed on the Kubernetes cluster that API Gateway is deployed on.
+- Versions of Kubernetes, `kubectl`, Consul and the Consul Helm chart are determined by the Consul API Gateway version deployed. Refer to [Release Notes](/docs/release-notes) for details.
```
Comment (nathancoleman) [inline website/content/docs/api-gateway/tech-specs.mdx:20]: I would have loved to link to `/docs/release-notes/api-gateway`; however, that's not an actual page and would require specifying a specific release in the URL

Comment (trujillo-adam) [reply website/content/docs/api-gateway/tech-specs.mdx:20]: @nathancoleman I think we need to keep this information here. The release notes should provide a history of the product changes, but the core docs should always contain requirements/technical specs. This is consistent with how other Consul add-ons document requirements.

**Review by trujillo-adam — CHANGES_REQUESTED** [review]
I understand the rationale for this change, but we should be consistent with how we document this kind of information across Consul add-ons. All other projects (Lambda, ECS, CTS) document requirements in the core docs and not in the release notes. I'm going to huddle with the rest of the Consul Ed team on this, though.

File: website/content/docs/api-gateway/tech-specs.mdx (hunk 1 above)
Comment (nathancoleman) [reply website/content/docs/api-gateway/tech-specs.mdx:20]: This makes sense. Will revert the most recent commit.

FYI @Jeff-Apple

**Review by trujillo-adam — APPROVED** [review]
👍🏽

**Pull request merged**